### ISSUE TITLE
Improve accessibility of sortable and filterable table

### ISSUE THREAD

**Issue by luistoptal** [gigascience/gigadb-website#1564]
## User story

Let's pick the table in this page as an example, and fix a11y issues for it http://gigadb.gigasciencejournal.com:9170/adminDataset/admin

- interactive header items must be distinguishable from non interactive header even in grayscale: optimal solution is to use button and down / up carets, simple solution is to underline text content
- Add hidden label (filter by ...) to filter inputs
- for screen readers it would be better if interactive table head links had role button (or be html buttons) instead of links
- table headers are simply announced by screen readers as "link". It should announce that it is for sorting. (There is a table that announces sorting in dataset details page)
- When clicking on sorting, focus is lost (acts as page refresh). Focus should remain

## Additional Info

Related to #1163 and #1155
